plugin/src/main/java/org/wildfly/plugin/provision/ApplicationImageMojo.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2022, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.plugin.provision;
+
+import static java.lang.String.format;
+import static java.lang.String.join;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "image", requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, defaultPhase = LifecyclePhase.PACKAGE)
+public class ApplicationImageMojo extends PackageServerMojo {
+
+    public static final int DOCKER_CMD_CHECK_TIMEOUT = 3000;
+    private static final String DOCKER_BINARY = "docker";
```

Comment:
Why not make this a parameter so that we can use podman if we want/need to ?

Reply:
As mentioned, this is a nice-to-have that can be considered. I can't run `podman` on my machine to test it right now.

Podman suggest to alias it to `docker` so this can be a workaround in the mean time.
If you test that it works with the alias, I could more confidently make the binary configurable too.

Reply:
I can do that :)

Reply:
done, you can configure it with `<dockerBinary>` (default is `docker`)